rotkehlchen/db/history_events.py

```diff
@@ -620,18 +625,33 @@ def get_history_events_count(
             cursor: 'DBCursor',
             query_filter: HistoryBaseEntryFilterQuery,
             group_by_event_ids: bool = False,
-    ) -> int:
-        """Returns how many events matching the filter but ignoring pagination are in the DB"""
-        prepared_query, bindings = query_filter.prepare(
-            with_pagination=False,
-            with_group_by=group_by_event_ids,
-        )
-        query = query_filter.get_count_query() + prepared_query
-        cursor.execute(query, bindings)
-        if group_by_event_ids:  # due to GROUP BY we get a list of all grouped by counts
-            return len(cursor.fetchall())
-        # else
-        return cursor.fetchone()[0]  # count(*) always returns
+            entries_limit: Optional[int] = None,
+    ) -> tuple[int, int]:
+        """
+        Returns how many events matching the filter but ignoring pagination are in the DB.
+        We return two integers the first one being the number of events returned and the second
+        the number of events if any limit is applied, otherwise the second value matches
+        the first.
+        """
+        prepared_query, bindings = query_filter.prepare(with_pagination=False)
+        # we need to select everything because any column could be used in the filter
+        query = 'SELECT * ' + query_filter.get_join_query() + prepared_query
+        if group_by_event_ids:
+            query = f'SELECT event_identifier FROM ({query}) GROUP BY event_identifier'
+        query = f'SELECT COUNT(*) FROM ({query})'
```

Comment:
Something is wrong here. You are assignign query in all 3 lines. What did you plan to do?